pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -221,6 +235,52 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         return _managementSwapFeePercentage;
     }
 
+    function getSwapFeePercentage() public view virtual override returns (uint256) {
+        uint256 pctProgress = _calculateGradualChangeProgress(_FEE_START_TIME_OFFSET, _FEE_END_TIME_OFFSET);
+
+        return _interpolateValue(super.getSwapFeePercentage(), _endSwapFeePercentage, pctProgress);
+    }
+
+    /**
+     * @dev Return start time, end time, and endWeights as an array.
+     * Current weights should be retrieved via `getNormalizedWeights()`.
+     */
+    function getGradualFeeUpdateParams()
+        external
+        view
+        returns (
+            uint256 startTime,
+            uint256 endTime,
+            uint256 endSwapFeePercentage
+        )
+    {
+        // Load current pool state from storage
+        bytes32 poolState = _getMiscData();
+
+        startTime = poolState.decodeUint32(_FEE_START_TIME_OFFSET);
+        endTime = poolState.decodeUint32(_FEE_END_TIME_OFFSET);
+        endSwapFeePercentage = _endSwapFeePercentage;
+    }
+
+    function _setSwapFeePercentage(uint256 swapFeePercentage) internal virtual override {
+        // Do not allow setting if there is an ongoing fee change
```

Comment:
Should we just overwrite the ongoing gradual weight change? A pool manager can currently just do `updateSwapFeeGradually(block.timestamp, block.timestamp, newSwapFeePercentage)` to instantly set the swap fee percentage to a new value.

Reply:
Well the intent here was to keep the interface and events consistent. setSwapFeePercentage is in the interface from BasePool, so if they call it, it should emit the normal event instead of the gradual change one. V1 also worked like this - to minimize the interactions (and test cases, and attack surface), it only let you do one thing at a time.

Reply:
Fair enough. Events are kinda tricky on this contract as you need to track both the instantaneous and gradual change events to get the current value of the swap fee.

I was just thinking on whether we can save the two error code slots but this seems reasonable.